client/my-sites/earn/components/add-edit-coupon-modal/test/products-selector.jsx

```diff
@@ -0,0 +1,210 @@
+	...jest.requireActual( '@wordpress/i18n' ),
+	__: jest.fn(),
+} ) );
+
+import { act, render, screen } from '@testing-library/react';
+import { __ } from '@wordpress/i18n';
+import { translate, useTranslate } from 'i18n-calypso';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import ProductsSelector from '../products-selector';
+
+const testProduct1 = {
+	ID: 9,
+	currency: 'USD',
+	price: 5,
+	title: 'Monthly Subscription',
+	interval: '1 month',
+	buyer_can_change_amount: false,
+	multiple_per_user: false,
+	welcome_email_content: 'Welcome!',
+	subscribe_as_site_subscriber: true,
+	renewal_schedule: '1 month',
+	type: 'newsletter',
+	is_editable: false,
+};
+
+const testProduct2 = {
+	ID: 10,
+	currency: 'USD',
+	price: 40,
+	title: 'Yearly Subscription',
+	interval: '1 year',
+	buyer_can_change_amount: false,
+	multiple_per_user: false,
+	welcome_email_content: 'Welcome!',
+	subscribe_as_site_subscriber: true,
+	renewal_schedule: '1 year',
+	type: 'newsletter',
+	is_editable: false,
+};
+
+const initialState = {
+	sites: { items: {} },
+	siteSettings: { items: {} },
+	ui: { selectedSiteId: 1 },
+	memberships: {
+		couponList: {
+			items: {},
+		},
+		productList: {
+			items: {
+				1: [ testProduct1, testProduct2 ],
+			},
+		},
+	},
+};
+
+function WrappedProductsSelector( props ) {
+	const mockStore = configureStore();
+	const store = mockStore( initialState );
+
+	return (
+		<Provider store={ store }>
+			<ProductsSelector { ...props } />
+		</Provider>
+	);
+}
+
+describe( 'ProductsSelector', () => {
+	const getAnyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', { name: 'Any product' } );
+	const getYearlyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', {
+			name: 'Yearly Subscription : $40.00 / year',
+		} );
+	const getMonthlyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', {
+			name: 'Monthly Subscription : $5.00 / month',
+		} );
+	const getButtonText = ( selectedCount ) => {
+		if ( selectedCount > 0 ) {
+			if ( selectedCount > 1 ) {
+				return selectedCount + ' products selected.';
+			}
+			return '1 product selected';
+		}
+		return 'Any product';
+	};
+	const getDropdownButton = ( selectedCount ) =>
+		screen.getByRole( 'button', { name: getButtonText( selectedCount ) } );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+
+		useTranslate.mockImplementation( mockUseTranslate );
+		translate.mockImplementation( mockTranslate );
+		__.mockImplementation( mockTranslate );
+	} );
+
+	test( 'should render with any product selected when no specific products have been selected', () => {
+		let editedPlanIdsAllowList = [];
+		const setEditedPlanIdsAllowList = ( list ) => {
+			editedPlanIdsAllowList = list;
+		};
+		render(
+			<WrappedProductsSelector
+				onSelectedPlanIdsChange={ ( list ) => setEditedPlanIdsAllowList( list ) }
+				initialSelectedList={ editedPlanIdsAllowList }
+				allowMultiple={ true }
```

Comment:
`={ true }` is redundant

```suggestion
				allowMultiple
```